be/src/vec/spill/spill_stream_manager.cpp

```diff
@@ -266,6 +267,20 @@ void SpillStreamManager::gc(int64_t max_file_count) {
     }
 }
 
+void SpillStreamManager::cleanup_query(TUniqueId query_id) {
+    for (auto& [_, store] : _spill_store_map) {
+        std::string query_spill_dir =
```

Comment:
This should not be a sync api. Because this is called in ~QueryContext, it may hang pipeline thread.
This API should be an async API, it only submit a task to spill thread pool. And the io thread will delete these folder async.

Reply:
It's a directory rename, it should not take much time.